File: services/trackbattle/src/back/models/comment.py
from sqlalchemy import Column, Integer, String, Date

from models.base import BaseModel


class Comment(BaseModel):
    __tablename__ = 'comments'
    id = Column(Integer, primary_key=True)
    track = Column(String)
    description = Column(String)
    likes_amount = Column(Integer)
    author_nickname = Column(String)
    post_id = Column(Integer)
    publishing_date = Column(Date)

    def __repr__(self):
        return "<Comment(id='{}', track='{}', description='{}', likes_amount='{}', author_nickname='{}', post_id='{}', publishing_date='{}'>" \
            .format(self.id,
                    self.track,
                    self.description,
                    self.likes_amount,
                    self.author_nickname,
                    self.post_id,
                    self.publishing_date)

File: services/trackbattle/src/front/src/api/types/Comment.ts
import {JsonObject, JsonProperty} from "json2typescript";

@JsonObject("Comment")
export class Comment {
    @JsonProperty("id", String, true)
    id: string = "";

    @JsonProperty("post_id", String, true)
    postId: string = "";

    @JsonProperty("author_nickname", String, false)
    author: string = "";

    @JsonProperty("publishing_date", String, false)
    publishingDate: string = "";

    @JsonProperty("description", String, true)
    description: string = "";

    @JsonProperty("likes_amount", Number, false)
    likes_amount: number = 0;

    @JsonProperty("track", String, false)
    track: string = "";
}

@JsonObject("CommentResponse")
export class CommentResponse {
    @JsonProperty("track", String, false)
    track: string = "";

    @JsonProperty("description", String, false)
    description: string = "";

    @JsonProperty("post_id", String, false)
    post_id: string = "";
}